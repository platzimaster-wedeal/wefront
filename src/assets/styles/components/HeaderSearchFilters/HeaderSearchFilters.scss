@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.header-search-filters {
  position: relative;
  width: 100%;

  &__title {
    @include font-size($font-size-medium);
  }

  &__inputs {
    display: flex;
    flex-flow: column nowrap;
    margin: $space-size-medium 0;

    & strong {
      @include font-size($font-size-normal);
    }
  }

  &__inputs--range {
    display: flex;
    flex-flow: column nowrap;
    margin: $space-size-small 0;

    & input[type="range"] {
      width: 80%;
      margin: $space-size-extra-small 0;
    }
  }

  &__prices {
    color: $color-primary;
  }

  &__select {
    min-width: 90px;
    margin: 0 $space-size-small;
  }
}
