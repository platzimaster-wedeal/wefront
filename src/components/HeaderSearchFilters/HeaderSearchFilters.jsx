import React, { useState } from "react";
import { useSelector } from "react-redux";
import { useInputForm } from "../../hooks/useInputForm/useInputForm";

// Styles
import "../../assets/styles/components/HeaderSearchFilters/HeaderSearchFilters.scss";

// Components
import SelectCategories from "../SelectCategories/SelectCategories";
import Button from "../Buttons/Button";

const HeaderSearchFilters = () => {
 const { categories } = useSelector((state) => state.CategoriesReducer);

 const [location, setLocation] = useInputForm("");
 const [minimumPrice, setMinimumPrice] = useInputForm(0);
 const [maximumPrice, setMaximumPrice] = useInputForm(0);
 const [filterCategories, setFilterCategories] = useState([]);

 const handleSubmit = (ev) => {
  ev.preventDefault();
 };

 return (
  <div className="header-search-filters">
   <strong className="header-search-filters__title">Filters</strong>

   <div className="header-search-filters__inputs">
    <strong>Location</strong>
    <label htmlFor="location">
     Current Country:
     <select
      id="location"
      onChange={setLocation}
      className="header-search-filters__select">
      <option value="USA">USA</option>
      <option value="COL">COL</option>
     </select>
    </label>
   </div>

   <div className="header-search-filters__inputs">
    <strong>Category</strong>
    <SelectCategories
     categories={categories}
     userCategories={filterCategories}
     setCategories={setFilterCategories}
     title="Select a category:"
    />
   </div>

   <div className="header-search-filters__inputs">
    <strong>Price</strong>
    <div className="header-search-filters__inputs--range">
     <label htmlFor="minimunPrice">
      Select a minimun price:{" "}
      <strong className="header-search-filters__prices">{minimumPrice}</strong>{" "}
     </label>
     <input
      type="range"
      id="minimunPrice"
      onChange={setMinimumPrice}
      min="0"
      max="1000"
      value={minimumPrice}
     />
    </div>
    <div className="header-search-filters__inputs--range">
     <label htmlFor="maximunPrice">
      Select a maximun price:{" "}
      <strong className="header-search-filters__prices">{maximumPrice}</strong>{" "}
     </label>
     <input
      type="range"
      id="maximunPrice"
      onChange={setMaximumPrice}
      min="0"
      max="1000"
      value={maximumPrice}
     />
    </div>
   </div>

   <Button
    onClick={handleSubmit}
    active
    className="header-search-filters__action">
    Apply
   </Button>
  </div>
 );
};

export default HeaderSearchFilters;
