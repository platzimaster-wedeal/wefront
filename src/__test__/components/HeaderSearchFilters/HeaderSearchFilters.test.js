import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import HeaderSearchFilters from "../../../components/HeaderSearchFilters/HeaderSearchFilters";

describe("Test for the component HeaderSearchFilters :D", () => {
 const HeaderSearchFiltersMounted = mount(<HeaderSearchFilters />);
 const HeaderSearchFiltersCreated = create(<HeaderSearchFilters />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(HeaderSearchFiltersCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(HeaderSearchFiltersMounted.length).toEqual(1);
  });

  test("Should the title of the component be a string", () => {
   expect(
    typeof HeaderSearchFiltersMounted.find(
     ".header-search-filters__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be Filters", () => {
   expect(
    HeaderSearchFiltersMounted.find(".header-search-filters__title").text()
   ).toEqual("Filters");
  });
 });
});
