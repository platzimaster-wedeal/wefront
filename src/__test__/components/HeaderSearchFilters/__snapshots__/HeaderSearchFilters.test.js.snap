// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component HeaderSearchFilters :D Visual tests for the component should the comopnent have the same tree 1`] = `
<form
  className="header-search-filters"
  onSubmit={[Function]}
>
  <strong
    className="header-search-filters__title"
  >
    Filters
  </strong>
  <div
    className="header-search-filters__inputs"
  >
    <strong>
      Location
    </strong>
    <label
      htmlFor="location"
    >
      Current Country:
      <select
        className="header-search-filters__select"
        id="location"
        onChange={[Function]}
      >
        <option
          value="USA"
        >
          USA
        </option>
        <option
          value="COL"
        >
          COL
        </option>
      </select>
    </label>
  </div>
  <div
    className="header-search-filters__inputs"
  >
    <strong>
      Category
    </strong>
    <div
      className="select-categories"
    >
      <label
        className="select-categories__title"
        htmlFor="categories"
      >
        Select a category:
      </label>
      <input
        id="categories"
        onChange={[Function]}
        placeholder="Search and select the skills that you want"
        type="text"
      />
      <ul
        className="select-categories__select-list"
      >
        <li
          onClick={[Function]}
        >
          <span
            className="label label__inactive"
          >
            Art
          </span>
        </li>
        <li
          onClick={[Function]}
        >
          <span
            className="label label__inactive"
          >
            Engineer
          </span>
        </li>
        <li
          onClick={[Function]}
        >
          <span
            className="label label__inactive"
          >
            AI
          </span>
        </li>
      </ul>
      <small
        className="select-categories__title"
      >
        Selected Options
      </small>
      <ul
        className="select-categories__list"
      >
        <li>
          There is no options selected.
        </li>
      </ul>
    </div>
  </div>
  <div
    className="header-search-filters__inputs"
  >
    <strong>
      Price
    </strong>
    <div
      className="header-search-filters__inputs--range"
    >
      <label
        htmlFor="minimunPrice"
      >
        Select a minimun price:
         
        <strong
          className="header-search-filters__prices"
        >
          0
        </strong>
         
      </label>
      <input
        id="minimunPrice"
        max="1000"
        min="0"
        onChange={[Function]}
        type="range"
        value={0}
      />
    </div>
    <div
      className="header-search-filters__inputs--range"
    >
      <label
        htmlFor="maximunPrice"
      >
        Select a maximun price:
         
        <strong
          className="header-search-filters__prices"
        >
          0
        </strong>
         
      </label>
      <input
        id="maximunPrice"
        max="1000"
        min="0"
        onChange={[Function]}
        type="range"
        value={0}
      />
    </div>
  </div>
  <button
    className="button button__primary header-search-filters__action"
    type="button"
  >
    Apply
  </button>
</form>
`;
